### PR TITLE
Create BLAST tabular adapter

### PR DESCRIPTION
**Pull request by garrettjstevens**
This adds a synteny adapter that works with files generated by BLAST usint the `-outfmt 6` option (called "tabular" output in the docs).

You can see a demo of this branch here: https://s3.amazonaws.com/jbrowse.org/code/jb2/create_blast_tabular_adapter/index.html?config=test_data%2Fgrape_peach_synteny%2Fconfig.json&session=share-4SDMC98yDL&password=JI48K

The demo file is just matches for the first 1,000,000 bases of Pp05 for peach and chr18 for grape. The file was generated like this:

```
# The files we have are Ppersica_Pp05_subset.fa and Vvinifera_chr18_subset.fa,
# both of with are the first 1,000,000 bases of the indicated chromosomes.
makeblastdb -in Vvinifera_chr18_subset.fa -dbtype nucl
tblastx -query Ppersica_Pp05_subset.fa -db Vvinifera_chr18_subset.fa -outfmt 6 -evalue 0.1 > peach_vs_grape.tsv
```

This took ~1m15s to generate on my computer.

### CODE REVIEW

File: plugins/comparative-adapters/src/BlastTabularAdapter/BlastTabularAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+
+  public static capabilities = ['getFeatures', 'getRefNames']
+
+  constructor(
+    public config: BlastTabularAdapterConfig,
+    public getSubAdapter?: getSubAdapterType,
+    public pluginManager?: PluginManager,
+  ) {
+    super(config, getSubAdapter, pluginManager)
+    this.data = this.setup(config)
```
Comment (cmdcolin) [inline plugins/comparative-adapters/src/BlastTabularAdapter/BlastTabularAdapter.ts:114]: I would avoid calling async operations in the constructor like this, it is difficult to provide error handling this way. you can possibly reference other ways that this has been done with "setupPre" for example, and then any function that needs to download the dataset calls another function "setup" that caches the setupPre promise

File: test_data/grape_peach_synteny/config.json [hunk 2]
```diff
@@ -52,6 +52,20 @@
         "assemblyNames": ["peach", "grape"]
       },
       "assemblyNames": ["peach", "grape"]
+    },
+    {
+      "type": "SyntenyTrack",
```
Comment (cmdcolin) [inline test_data/grape_peach_synteny/config.json:57]: might be good to add the GFF track for this so you can see the features line up

Comment (cmdcolin) [reply test_data/grape_peach_synteny/config.json:57]: i tried adding e.g.

```
{
      "type": "FeatureTrack",
      "trackId": "grape_genes",
      "name": "Vvinifera_457_v2.1.gene.sorted.gff",
      "adapter": {
        "type": "Gff3TabixAdapter",
        "gffGzLocation": {
          "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/grape/Vvinifera_457_v2.1.gene.sorted.gff.gz"
        },
        "index": {
          "location": {
            "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/grape/Vvinifera_457_v2.1.gene.sorted.gff.gz.tbi"
          }
        }
      },
      "assemblyNames": ["grape"]
    },
    {
      "type": "FeatureTrack",
      "trackId": "peach_genes",
      "name": "Ppersica_298_v2.1.gene.sorted.gff3",
      "adapter": {
        "type": "Gff3TabixAdapter",
        "gffGzLocation": {
          "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/peach/Ppersica_298_v2.1.gene.sorted.gff3.gz"
        },
        "index": {
          "location": {
            "uri": "https://s3.amazonaws.com/jbrowse.org/genomes/peach/Ppersica_298_v2.1.gene.sorted.gff3.gz.tbi"
          }
        }
      },
      "assemblyNames": ["peach"]
    },
```

but nothing was lining up

**Comment by cmdcolin** [comment]
I rebased this off main and gzipped the sample data file  in case you get merge conflicts when pulling!

File: plugins/comparative-adapters/src/BlastTabularAdapter/BlastTabularAdapter.ts (hunk 1 above)
Comment (garrettjstevens) [reply plugins/comparative-adapters/src/BlastTabularAdapter/BlastTabularAdapter.ts]: Fixed

File: test_data/grape_peach_synteny/config.json (hunk 2 above)
Comment (garrettjstevens) [reply test_data/grape_peach_synteny/config.json:91]: I've added these tracks

**Comment by garrettjstevens** [comment]
Here's a new share link with the gene tracks: https://s3.amazonaws.com/jbrowse.org/code/jb2/create_blast_tabular_adapter/index.html?config=test_data%2Fgrape_peach_synteny%2Fconfig.json&session=share-csnO59YS7_&password=wSNlc

image

I also added a config option so users can specify their columns in case they used the custom column option in BLAST's outfmt. If you ran the command with `-outfmt "6 qseqid sseqid qstart qend sstart send"`, then you would specify the columns in the config as `'qseqid sseqid qstart qend sstart send'`. Those six columns are required, all others are optional.

**Comment by cmdcolin** [comment]
if you get a chance, i would be curious how outfmt 17 (SAM, then convert to BAM) works in jbrowse too...

**Comment by garrettjstevens** [comment]
outfmt 17 only works with blastn, so I've attached both the outfmt 6 file and outfmt 17 file using blastn. For some reason the raw SAM output has `Query_1` instead of `PP05`, so I replaced it and converted to BAM, and those files are attached, too.

Here's what it looks like with outfmt 6 loaded as a synteny track and outfmt 17 loaded as an alignments track:

image

[peach_vs_grape.blastn.tar.gz](https://github.com/user-attachments/files/17682771/peach_vs_grape.blastn.tar.gz)

**Pull request merged**